packages/fetchai/skills/simple_oracle/behaviours.py

```diff
@@ -107,15 +107,19 @@ def act(self) -> None:
             self._request_grant_role_transaction()
             return
 
-        # Check for observation from data collecting skill
-        observation = self.context.shared_state.get("observation", None)
-        if observation is None:
+        # Check for oracle from data collecting skill
+        if strategy.aggregation:  # pragma: nocover
+            oracle_data = self.context.shared_state.get("aggregation", None)
+        else:
+            oracle_data = self.context.shared_state.get("observation", None)
```

Comment:
see above for both keys

Reply:
Not sure why we need the if/else here. Difference is just the key, once that's configurable you can remove it.